homeassistant/components/tplink/light.py

```diff
@@ -214,6 +213,11 @@ def __init__(
             # If the light supports only a single color mode, set it now
             self._fixed_color_mode = next(iter(self._attr_supported_color_modes))
 
+        # We enable transitions only for kasa devices.
+        # Use credentials as proxy for tapo device detection.
+        if device.credentials is None:
```

Comment:
I think it would be better to implement this in the library similar to how it's done for the [`energy` interface](https://github.com/python-kasa/python-kasa/blob/master/kasa/interfaces/energy.py#L17) rather than using credentials as a proxy.

Also if we are going to comment on device types probably better to refer to them as the newer `smart` models as this would affect kasa products as well, e.g. ks240.

Reply:
Agreed, it'd be much nicer to have some way to signal about supported (module) features :+1: